File: spring2013/codeWar/LRS/Clients/PlayerCSharpAI/api/Laser.cs
// Created by Windward Studios, Inc. (www.windward.net). No copyright claimed - do anything you want with this code.

using System.Xml.Linq;

namespace PlayerCSharpAI.api
{
	/// <summary>
	/// A laser on the board. This sits on a specific map square.
	/// </summary>
	public class Laser : CombatUnit
	{

		/// <summary>
		/// Copy constructor.
		/// </summary>
		/// <param name="element">Initialize with the values in this XML.</param>
		public Laser(XElement element)
			: base(element)
		{
			NumSquares = int.Parse(element.Attribute("num-squares").Value);
		}

		/// <summary>
		/// The number of squares the laser shoots over, including the one it is on. (The Laser is ended by a wall.)
		/// </summary>
		public int NumSquares { get; private set; }
	}
}

File: spring2013/codeWar/LRS/Clients/PlayerCppAI/Laser.cpp
#include "Laser.h"

namespace PlayerCSharpAI
{
	namespace api
	{
		Laser::Laser() : CombatUnit()
		{

		}

		Laser::Laser(const xml_node& element) : CombatUnit(element)
		{
			m_numSquares = element.attribute("num-squares").as_int();
		}

		Laser::Laser(const Laser& l) : CombatUnit(l)
		{
			m_numSquares = l.m_numSquares;
		}

		int Laser::getNumSquares() const
		{
			return m_numSquares;
		}

		Laser& Laser::operator=(const Laser& l)
		{
			CombatUnit::operator=(l);
			m_numSquares = l.m_numSquares;
			return *this;
		}

		/*void Laser::setNumSquares(int value)
		{
			m_numSquares = value;
		}*/
	}
}
